Source/WebCore/platform/graphics/skia/GraphicsContextSkia.cpp

```diff
@@ -953,7 +1003,7 @@ void GraphicsContextSkia::drawPattern(NativeImage& nativeImage, const FloatRect&
     if (!patternTransform.isInvertible())
         return;
 
-    auto image = nativeImage.platformImage();
+    auto image = skiaImageFromNativeImage(*this, nativeImage);
```

Comment:
I wonder if we always want platformImage to return a valid image on any thread instead.

Reply:
That will be empty when painting it though...

Reply:
What do you mean? What will be empty?

Reply:
If you attempt to use the image without re-wrapping the SkSurface the painted content is empty (nothing visible)

Reply:
Hmm good question. If native image would directly execute the wrapping logic, it would be very inefficient to call platformImage() multiple times in a row.

I will check all call sites and check if this is feasible...

Reply:
If that's not the case I would still move the code to NativeImage with a better name, because right now NativeImage::platformImage and skiaImageFromNativeImage are the same if you just look at the name, it's not obvios what to use. That's why I suggested NativeImage::crossThreadPlatformImage